ui/loading_indicators.py

```diff
@@ -1,59 +1,182 @@
-from ui import Refresher
-from ui.utils import clamp
+from __future__ import division
+
+import math
+from collections import namedtuple
+from math import cos
 from threading import Thread
+from time import time
 
+from PIL import ImageDraw
+from luma.core.render import canvas
 
-class LoadingIndicator(Refresher):
+from ui import Refresher
+from ui.utils import clamp, Counter, to_be_foreground
+
+"""
+These classes subclass `Refresher` to show an animated screen.
+Those screens are used to show the user something is happening in the background.
+
+There are two possibility : either you know how much work is happening in the background, or you don't.
+
+
+"""
 
+Rect = namedtuple('Rect', ['left', 'top', 'right', 'bottom'])
+
+
+class CenteredTextRenderer(object):
+    def draw_centered_text(self, draw, content, device_size):
+        # type: (ImageDraw, str, tuple) -> None
+        # draws a centered text on the canvas and returns a 4-tuple of the coordinates taken by the text
+        coords = self.get_centered_text_bounds(draw, content, device_size)
+        draw.text((coords.left, coords.top), content, fill=True)
+
+    @staticmethod
+    def get_centered_text_bounds(draw, content, device_size):
+        # type: (ImageDraw, str, tuple) -> Rect
+        # returns the coordinates of the centered text (min_x, min_y, max_x, max_y)
+        w, h = draw.textsize(content)
+        dw, dh = device_size
+        return Rect(dw / 2 - w / 2, dh / 2 - h / 2, dw / 2 + w / 2, dh / 2 + h / 2)
+
+
+class LoadingIndicator(Refresher):
+    """
+    Abstract class to indicate a loading time to the user
+    """
     def __init__(self, i, o, *args, **kwargs):
+        Refresher.__init__(self, self.on_refresh, i, o)
```

Comment:
You still have the duplicate `__init__` call - sorry, should've explicitly pointed at it, too

Reply:
fixed, thanks